### PR TITLE
Fix custom endpoints register and other issues

### PR DESCRIPTION
**Pull request by pedromcp90**
## Required-by:
- https://github.com/doofinder/support/issues/2102

## What's changed:
- Fixed an issue while registering the custom endpoints.
- Added some checks in Thumbnails class to avoid warnings.
- Ignore invalid WC products to avoid indexing invalid products.

### CODE REVIEW

File: doofinder-for-woocommerce/includes/class-thumbnail.php [hunk 1]
```diff
@@ -74,7 +74,8 @@ public function get()
 	 * @param string $thumbnail_id Thumb ID por check
 	 * @return string URL of Thumb or false
 	 */
-	private function get_image_url_by_size($thumbnail_id){
+	private function get_image_url_by_size($thumbnail_id)
```
Comment (davidmolinacano) [inline doofinder-for-woocommerce/includes/class-thumbnail.php:77]: I think this is caused by automatic formatting, but this new format goes against WP PHP coding standards: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#brace-style

File: doofinder-for-woocommerce/doofinder-for-woocommerce.php [hunk 2]
```diff
@@ -354,8 +352,20 @@ public static function register_notices_styles()
             wp_enqueue_style( 'doofinder-notice', Doofinder_For_WordPress::plugin_url() . '/assets/css/doofinder-notice.css', array(), self::$version );
         }
 
+        /**
+         * This method initializes REST API endpoints.
+         * 
+         * We must remember that 'rest_api_init' hooks should be added outside the 'init' hook to prevent endpoints from
+         * not being registered because the 'rest_api_init' hook is executed earlier. 
+         * This is because we cannot guarantee that the order will always be 'init > rest_api_init'.
```
Comment (davidmolinacano) [inline doofinder-for-woocommerce/doofinder-for-woocommerce.php:360]: Excellent explanation!!! :clap: :trophy:

**Review by davidmolinacano — APPROVED** [review]
LGTM, just a small optional suggestion :clap: :clap:

File: doofinder-for-woocommerce/includes/class-thumbnail.php (hunk 1 above)
Comment (pedromcp90) [reply doofinder-for-woocommerce/includes/class-thumbnail.php:77]: I'm sorry, my IDE is configured to work with PSR Standards, I'll change the configuration to adapt to the WordPress standards and update the code.

**Review by eduardogomez97 — APPROVED** [review]

**Pull request merged**